src/backend/version-tracker/version-tracker.lambda.ts

```diff
@@ -0,0 +1,167 @@
+  for (const [idx, batch] of requestBatches.entries()) {
+    console.log(`Batch ${idx} of ${requestBatches.length}`);
+
+    const promises = batch.map(async (packagePrefix) => {
+      const versionPrefixes = await listPrefixes(PACKAGE_DATA_BUCKET_NAME, packagePrefix);
+      if (versionPrefixes.length === 0) return;
+      const [, name] = PACKAGE_PREFIX_REGEX.exec(versionPrefixes[0])!;
+      const versions = new Array<string>();
+      for (const versionPrefix of versionPrefixes) {
+        const [,, version] = PACKAGE_PREFIX_REGEX.exec(versionPrefix)!;
+        versions.push(version);
+      }
+      versionMap.set(name, versions);
+    });
+
+    await Promise.all(promises);
+  }
+
+  const versionJson = {
+    packages: Object.fromEntries(versionMap),
+    lastUpdated: new Date().toISOString(),
+  };
+
+  let totalVersions = 0;
+  versionMap.forEach((versions) => { totalVersions += versions.length; });
+
+  // Update metrics.
+  console.log(`${versionMap.size} package versions have been recorded.`);
+  await metricScope((metrics) => async () => {
+    // Clear out default dimensions as we don't need those. See https://github.com/awslabs/aws-embedded-metrics-node/issues/73
+    metrics.setDimensions();
+
+    metrics.setNamespace(METRICS_NAMESPACE);
+    metrics.putMetric(MetricName.TRACKED_PACKAGES_COUNT, versionMap.size, Unit.Count);
+    metrics.putMetric(MetricName.TRACKED_VERSIONS_COUNT, totalVersions, Unit.Count);
+  })();
+
+  // Upload the result to S3 and exit.
+  const result = await aws.s3().putObject({
+    Bucket: VERSION_TRACKER_BUCKET_NAME,
+    Key: VERSION_TRACKER_OBJECT_KEY,
+    Body: JSON.stringify(versionJson),
+    ContentType: 'application/json',
+    CacheControl: 'public, max-age=60, must-revalidate, proxy-revalidate', // Expire from cache after 1 minute
+    Metadata: {
+      'Lambda-Log-Group': context.logGroupName,
+      'Lambda-Log-Stream': context.logStreamName,
+      'Lambda-Run-Id': context.awsRequestId,
+      'Package-Count': `${versionMap.size}`,
+      'Version-Count': `${totalVersions}`,
+    },
+  }).promise();
+
+  return result;
+}
+
+/**
+ * List all prefixes in a bucket given a base prefix.
+ *
+ * For example, if given "data/" it will return
+ * [
+ *   "data/@aws-amplify/",
+ *   "data/@aws-c2a/",
+ *   "data/@aws-cdk/",
+ *   ...
+ * ]
+ */
+async function listPrefixes(bucket: string, prefix: string): Promise<string[]> {
+  const prefixes = new Array<string>();
+
+  let continuationToken;
+  do {
+    const listRequest: AWS.S3.ListObjectsV2Request = {
+      Bucket: bucket,
+      Prefix: prefix,
+      Delimiter: '/',
+      ContinuationToken: continuationToken,
+    };
+    const listResponse = await aws.s3().listObjectsV2(listRequest).promise();
+    continuationToken = listResponse.NextContinuationToken;
+
+    for (const { Prefix: commonPrefix } of listResponse.CommonPrefixes ?? []) {
+      if (!commonPrefix) { continue; }
+      prefixes.push(commonPrefix);
+    }
```

Comment:
Alternatively (granted you find it as/more expressive)

```suggestion
    prefixes.push(
      ...(listResponse.CommonPrefixes ?? [])
        .map(({ Prefix }) => Prefix)
        .filter((prefix) => !!prefix)
    );
```